openvdb_cmd/vdb_tool/include/Parser.h

```diff
@@ -0,0 +1,1041 @@
+    std::vector<Option>    options;// e.g. {{"grids", "density,sphere"}, {"files", "path/base.ext"}}
+    std::function<void()>  init, run;// callback functions
+
+    /// @brief c-tor
+    Action(std::string _name,
+           std::string _alias,
+           std::string _doc,
+           std::vector<Option> &&_options,
+           std::function<void()> &&_init,
+           std::function<void()> &&_run,
+           size_t _anonymous = -1)
+      : name(std::move(_name))
+      , alias(std::move(_alias))
+      , documentation(std::move(_doc))
+      , anonymous(_anonymous)
+      , options(std::move(_options))
+      , init(std::move(_init))
+      , run(std::move(_run)) {}
+    /// @brief default copy constructor
+    Action(const Action&) = default;
+    /// @brief Sets the options of this actions
+    void setOption(const std::string &str);
+    void print(std::ostream& os = std::cerr) const;
+};// Action struct
+
+// ==============================================================================================================
+
+using ActListT = std::list<Action>;
+using ActIterT = typename ActListT::iterator;
+using VecF = std::vector<float>;// vector of floats
+using VecI = std::vector<int>;// vector for integers
+using VecS = std::vector<std::string>;// vector of strings
+
+// ==============================================================================================================
+
+/// @brief Class that stores values by name
+class Memory
+{
+    std::unordered_map<std::string, std::string> mData;
+public:
+    Memory() = default;
+    std::string get(const std::string &name) {
+        auto it = mData.find(name);
+        if (it == mData.end()) {
+            if (name=="pi") {
+                return std::to_string(std::atan(1)*4);
+            } else if (name=="e") {
+                return std::to_string(2.718281828459);
+            } else {
+                throw std::invalid_argument("Storrage::get: undefined variable \""+name+"\"");
+            }
+        }
+        return it->second;
+    }
+    void clear() {mData.clear();}
+    void clear(const std::string &name) {mData.erase(name);}
+    void set(const std::string &name, const std::string &value) {mData[name]=value;}
+    void set(const std::string &name, const char *value) {mData[name]=value;}
+    template <typename T>
+    void set(const std::string &name, const T &value) {mData[name]=std::to_string(value);}
+    void print(std::ostream& os = std::cerr) const {
+        std::map<std::string, std::string> tmp(mData.begin(),mData.end());// sort output
+        for (auto &d : tmp) os << d.first <<"="<<d.second<<std::endl;
+    }
+    size_t size() const {return mData.size();}
+    bool isSet(const std::string &name) const {return mData.find(name)!=mData.end();}
+};// Memory
+
+// ==============================================================================================================
+
+class Stack {
+    std::vector<std::string> mData;
+public:
```

Comment:
As a general comment, the member variables are usually included at the end of the class definition.